### PR TITLE
Feature/uniqueness row level results

### PR DESCRIPTION
**Pull request by eycho-am**
*Issue #, if available:*
N/A

*Description of changes:*
Building on this [PR](https://github.com/awslabs/deequ/pull/451) to add more analyzers that support row level results.

This PR adds row level results for the `Uniqueness` analyzer, which is used by several checks (`isUnique`, `isPrimaryKey`, `hasUniqueness`, `uniqueValueRatio`)

Row level results for other `ScanShareableFrequencyBasedAnalyzers` will be added in a future PR.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by eycho-am** [comment]
Note: Tests for `FileSystemMetricsRepositoryTest` are currently failing. I'm working on a way to not serialize the `fullColumn` field of metrics, which is currently causing the issue:

```
Syntax Error
== SQL ==
CASE WHEN (count(att1) OVER (PARTITION BY att1, att2 ORDER BY att1 ASC NULLS FIRST unspecifiedframe$()) = 1) THEN true WHEN (count(att1) OVER (PARTITION BY att1, att2 ORDER BY att1 ASC NULLS FIRST unspecifiedframe$()) = 0) THEN NULL ELSE false END
        ----------^^^
```

**Comment by eycho-am** [comment]
> Note: Tests for `FileSystemMetricsRepositoryTest` are currently failing. I'm working on a way to not serialize the `fullColumn` field of metrics, which is currently causing the issue:
> 
> ```
> Syntax Error
> == SQL ==
> CASE WHEN (count(att1) OVER (PARTITION BY att1, att2 ORDER BY att1 ASC NULLS FIRST unspecifiedframe$()) = 1) THEN true WHEN (count(att1) OVER (PARTITION BY att1, att2 ORDER BY att1 ASC NULLS FIRST unspecifiedframe$()) = 0) THEN NULL ELSE false END
>         ----------^^^
> ```

Resolved this issue by not serializing the full column field in metrics for now

File: src/main/scala/com/amazon/deequ/metrics/Metric.scala [hunk 1]
```diff
@@ -69,6 +71,10 @@ case class DoubleMetric(
   extends Metric[Double] with FullColumn {
 
   override def flatten(): Seq[DoubleMetric] = Seq(this)
+
+  override def getMetricWithoutFullColumn: DoubleMetric = {
```
Comment (rdsharma26) [inline src/main/scala/com/amazon/deequ/metrics/Metric.scala:75]: Is this still required?

File: src/main/scala/com/amazon/deequ/repository/AnalysisResultSerde.scala [hunk 2]
```diff
@@ -513,7 +513,7 @@ private[deequ] object MetricSerializer extends JsonSerializer[Metric[_]] {
         result.addProperty("instance", doubleMetric.instance)
         result.addProperty("name", doubleMetric.name)
         result.addProperty("value", doubleMetric.value.getOrElse(null).asInstanceOf[Double])
-        doubleMetric.fullColumn.foreach(c => result.addProperty("fullColumn", c.expr.sql))
+        // doubleMetric.fullColumn.foreach(c => result.addProperty("fullColumn", c.expr.sql))
```
Comment (rdsharma26) [inline src/main/scala/com/amazon/deequ/repository/AnalysisResultSerde.scala:516]: Maybe we can create an issue in Github as well and link it here?

File: src/main/scala/com/amazon/deequ/repository/fs/FileSystemMetricsRepository.scala [hunk 3]
```diff
@@ -40,6 +47,7 @@ class FileSystemMetricsRepository(session: SparkSession, path: String) extends M
     */
   override def save(resultKey: ResultKey, analyzerContext: AnalyzerContext): Unit = {
 
+
```
Comment (rdsharma26) [inline src/main/scala/com/amazon/deequ/repository/fs/FileSystemMetricsRepository.scala:50]: excess whitespace

File: src/main/scala/com/amazon/deequ/repository/AnalysisResultSerde.scala (hunk 2 above)
Comment (eycho-am) [reply src/main/scala/com/amazon/deequ/repository/AnalysisResultSerde.scala:516]: done

File: src/main/scala/com/amazon/deequ/metrics/Metric.scala (hunk 1 above)
Comment (eycho-am) [reply src/main/scala/com/amazon/deequ/metrics/Metric.scala]: nope, removed

**Review by rdsharma26 — APPROVED** [review]

**Pull request merged**